test/test_constraints_file.py

```diff
@@ -230,3 +230,47 @@ def test_verify_working_days_for_employee_does_not_exceed_the_max_working_days()
                 employees_shifts[employee.employee_id] += 1
 
     assert (max(employees_shifts.values()) <= max_working_days)
+
+
+def test_no_optimal_solution_when_the_closing_shift_and_the_next_shift_are_too_close_to_each_other():
+    minimum_time_between_shifts = datetime.timedelta(hours=9)
+    test_employee = Employee("test", EmployeePriorityEnum.HIGHEST, EmployeeStatusEnum.senior_employee, employee_id=uuid4())
+
+    closing_shift = Shift(shift_id=uuid4(), shift_type=ShiftTypesEnum.CLOSING, start_time=datetime.datetime(2023, 12, 12, 16, 0), end_time=datetime.datetime(2023, 12, 13, 2, 0))
+    shift_too_close_to_closing_shift = Shift(shift_id=uuid4(), shift_type=ShiftTypesEnum.MORNING, start_time=closing_shift.end_time + minimum_time_between_shifts - datetime.timedelta(hours=1), end_time=datetime.datetime(2023, 12, 13, 22, 0))
+
+    shifts = [closing_shift, shift_too_close_to_closing_shift]
+    employees = [test_employee]
+
+    model = cp_model.CpModel()
+    all_shifts = generate_shift_employee_combinations(employees, shifts, model)
+
+    add_exactly_one_employee_per_shift_constraint(shifts, employees, model, all_shifts)
+    add_at_most_one_shift_per_employee_in_the_same_day_constraint(shifts, employees, model, all_shifts)
+    add_minimum_time_between_closing_shift_and_next_shift_constraint(shifts, employees, model, all_shifts, minimum_time_between_shifts)
+
+    solver = cp_model.CpSolver()
+    status = solver.Solve(model)
+    assert (status != cp_model.OPTIMAL)
+
+
+def test_every_employee_that_worked_closing_shift_does_not_work_the_next_shifts_that_are_too_close_to_the_closing_shift():
+    minimum_time_between_shifts = datetime.timedelta(hours=9)
+    test_employee = Employee("test", EmployeePriorityEnum.HIGHEST, EmployeeStatusEnum.senior_employee, employee_id=uuid4())
+
+    closing_shift = Shift(shift_id=uuid4(), shift_type=ShiftTypesEnum.CLOSING, start_time=datetime.datetime(2023, 12, 12, 16, 0), end_time=datetime.datetime(2023, 12, 13, 2, 0))
+    available_shift_after_closing_shift = Shift(shift_id=uuid4(), shift_type=ShiftTypesEnum.MORNING, start_time=closing_shift.end_time + minimum_time_between_shifts + datetime.timedelta(hours=1), end_time=datetime.datetime(2023, 12, 13, 22, 0))
```

Comment:
hardcoding the endtime makes the problem remain. create the `start_time` outside and the `end_time` can be set to x hours after that